### PR TITLE
Add upgrade flags to pip installs

### PR DESCRIPTION
**Pull request by dakinggg**
# What does this PR do?
Adds `--upgrade` to the `pip install`s in CI. This ensures that CI will flag a dependency that upgrade that breaks composer

Closes [CO-1707](https://mosaicml.atlassian.net/browse/CO-1707)

[CO-1707]: https://mosaicml.atlassian.net/browse/CO-1707?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/pytest-cpu.yaml [hunk 1]
```diff
@@ -48,7 +48,7 @@ jobs:
           set -ex
           export PATH=/composer-python:$PATH
           python -m pip install --upgrade pip wheel
-          python -m pip install -e .[all]
+          python -m pip install --upgrade -e .[all]
```
Comment (mvpatel2000) [inline .github/workflows/pytest-cpu.yaml:51]:
```suggestion
          python -m pip install --upgrade .[all]
```

File: .github/workflows/code-quality.yaml [hunk 2]
```diff
@@ -40,7 +40,7 @@ jobs:
         run: |
           set -ex
           python -m pip install --upgrade pip wheel
-          python -m pip install -e .${{ matrix.pip_deps }}
+          python -m pip install --upgrade -e .${{ matrix.pip_deps }}
```
Comment (mvpatel2000) [inline .github/workflows/code-quality.yaml:43]:
```suggestion
          python -m pip install --upgrade .${{ matrix.pip_deps }}
```

**Review by mvpatel2000 — COMMENTED** [review]
There's a few more hanging `-e` that I added when copy pasting things which shouldn't really be there (since we install after checking out the right branch). Can you remove them in this PR? 

Will approve after that change + merging with changes to `mcp_pytest`

File: .github/workflows/pytest-daily.yaml [hunk 3]
```diff
@@ -71,7 +71,7 @@ jobs:
           set -ex
           export PATH=/composer-python:$PATH
           python -m pip install --upgrade pip wheel
-          python -m pip install -e .[all]
+          python -m pip install --upgrade -e .[all]
```
Comment (mvpatel2000) [inline .github/workflows/pytest-daily.yaml:74]:
```suggestion
          python -m pip install --upgrade .[all]
```

**Review by mvpatel2000 — APPROVED** [review]
LGTM. Can you remove the `-e` I left in

File: .github/workflows/pytest-daily.yaml (hunk 3 above)
Comment (dakinggg) [reply .github/workflows/pytest-daily.yaml:74]: oops, i meant to do that. my fingers just typed something else.

**Pull request merged**